### ISSUE TITLE
When I was training a CNN+GRU model with CTC loss, I got the nan loss after several batches.

### ISSUE THREAD

**Issue by jiayi-wei** [pytorch/pytorch#3363]
My model:
`class CTCModel(nn.Module):
    def __init__(self,output_size,rnn_hidden_size=128, num_rnn_layers=1, dropout=0):
        super(CTCModel, self).__init__()
        self.num_rnn_layers = num_rnn_layers
        self.rnn_hidden_size = rnn_hidden_size
        self.output_size = output_size
        self.layer1 = nn.Sequential(
            #ResBlock(3,32),
            nn.Conv2d(3,32, kernel_size=(3,3),stride=(1,1),padding=(1,1)),
            nn.ReLU(),
            nn.Conv2d(32, 32, kernel_size=(3, 4), stride=(3, 2)),
            nn.BatchNorm2d(32),
            nn.ReLU(),
            #nn.Dropout2d(dropout)
        )
        self.layer2 = nn.Sequential(
            #ResBlock(32,32),
            nn.Conv2d(32,32, kernel_size=(3,3),stride=(1,1),padding=(1,1)),
            nn.ReLU(),
            nn.Conv2d(32, 32, kernel_size=(4, 3), stride=(4, 2)),
            nn.BatchNorm2d(32),
            nn.ReLU(),
            #nn.Dropout2d(dropout)
        )
        self.layer3 = nn.Sequential(
            #ResBlock(32,32),
            nn.Conv2d(32, 32, kernel_size=(3, 3),stride=(1,1),padding=(1,1)),
            nn.ReLU(),
            nn.Conv2d(32, 32, kernel_size=(4, 2), stride=(2, 2)),
            #nn.BatchNorm2d(32),
            nn.ReLU(),
            nn.Conv2d(32, 32, kernel_size=(3, 2), stride=(1, 1)),
            nn.BatchNorm2d(32),
			nn.ReLU(),
        )
        self.gru = nn.GRU(32, rnn_hidden_size, num_rnn_layers,
                          batch_first=True,
                          dropout=dropout,bidirectional=True)
        self.linear = nn.Linear(rnn_hidden_size*2,output_size)

    def forward(self, x, hidden):
        h0 = hidden
        #print h0.size()
        out = self.layer1(x)
        #print out.size()
        out = self.layer2(out)
        #print out.size()
        out = self.layer3(out).squeeze()
        #print out.size()
        out = out.transpose(1, 2)
        #print out.size()
        #quit()
        out, hidden = self.gru(out, h0)
        out = self.linear(out)
        return out

    def initHidden(self,batch_size,use_cuda=False):
        h0 = Variable(torch.zeros(self.num_rnn_layers*2,batch_size,self.rnn_hidden_size))
        if use_cuda:
            return (h0.cuda())
        else:
            return h0`

Training module:
`def CTCtrain(inputs,targets,lens,ctc,ctc_optimizer,criterion,clip,use_cuda=False):
    if use_cuda:
        inputs = inputs.cuda()
    loss = 0
    ctc_optimizer.zero_grad()
    batch_size = inputs.size()[0]
    init_hidden = ctc.initHidden(batch_size,use_cuda=use_cuda)
    ctc_outputs = ctc(inputs,init_hidden)
    #print ctc_outputs.size(), inputs.size(), init_hidden.size()
    #quit() 
    ctcloss_inputs = ctc_outputs.transpose(0,1) #SeqLen * BatchSize * Hidden
    label_lens = lens
    act_lens = Variable(torch.IntTensor(batch_size*[ctc_outputs.size()[1]]),requires_grad=False)
    #print ctcloss_inputs, targets, act_lens, label_lens
    #quit()
    loss = criterion(ctcloss_inputs,targets,act_lens,label_lens)

    loss.backward()
    torch.nn.utils.clip_grad_norm(ctc.parameters(), clip)
    ctc_optimizer.step()

    #TODO
    decoded_outputs = decode_ctc_outputs(ctc_outputs)
    decoded_targets = np.split(targets.data.numpy(),lens.data.numpy().cumsum())[:-1]
    accuracy = np.array([np.array_equal(decoded_targets[i],decoded_outputs[i])
                         for i in range(batch_size)]).mean()

    return loss.data[0],accuracy`

The size of input image is 100*300. The CTCLoss could update the weight for a while. Then, the loss while become nan

Could someone tell me the reason? THX!!

**Comment by apaszke** [comment]
We're using GitHub issues for bug reports only, and all questions should be posted on [our forums](discuss.pytorch.org).
